src/localplayer.cpp

```diff
@@ -1077,3 +1074,67 @@ float LocalPlayer::getSlipFactor(Environment *env, const v3f &speedH)
 	}
 	return 1.0f;
 }
+
+void LocalPlayer::handleAutojump(f32 dtime, Environment *env,
+		const collisionMoveResult &result, const v3f &old_position,
+		const v3f &old_speed, f32 pos_max_d)
+{
+	PlayerSettings &player_settings = getPlayerSettings();
+	bool control_forward = control.up ||
+			       (!control.up && !control.down &&
+					       control.forw_move_joystick_axis < -0.05);
+	if (autojump) {
+		// release autojump after a given time
+		autojump_time -= dtime;
+		if (autojump_time <= 0.0f) {
+			autojump = false;
+		}
+	} else if (m_can_jump && !control.jump && !control.sneak && control_forward &&
+			player_settings.autojump) {
+
+		bool horizontal_collision = false;
+		for (const auto &colinfo : result.collisions) {
+			if (colinfo.type == COLLISION_NODE && colinfo.plane != 1)
+				horizontal_collision = true;
+		}
+
+		if (horizontal_collision) {
+
+			// first check for nodes above
+			float collisionHeight = m_collisionbox.MaxEdge.Y;
+			v3f headpos_f = m_position + v3f(0.0f, collisionHeight, 0.0f);
+			v3s16 ceilpos = floatToInt(headpos_f, BS) + v3s16(0, 1, 0);
+			Map *map = &env->getMap();
```

Comment:
Why do you need a pointer here? Use the Map reference.

Reply:
Copy&paste. I've replaced it with a reference, in fact I've removed the `map` variable altogether (it was only used once).